src/coreclr/src/jit/importer.cpp

```diff
@@ -4015,6 +4015,45 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Type_get_IsClass:
+            case NI_System_Type_get_IsPrimitive:
+            case NI_System_Type_get_IsValueType:
+            {
+                if (impStackTop().val->IsCall())
+                {
+                    GenTreeCall* call = impStackTop().val->AsCall();
+                    if (call->gtCallMethHnd == eeFindHelper(CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE))
+                    {
+                        GenTreeIntCon* handle = call->Args().begin()->GetNode()->AsIntCon();
+                        auto hClass           = reinterpret_cast<CORINFO_CLASS_HANDLE>(handle->IconValue());
+                        typeInfo tinfo        = verMakeTypeInfo(hClass);
```

Comment:
I do not think you want to use `verMakeTypeInfo` here. It won't do the right thing for `ldtoken int&` according to the comments in the implementation. You cannot do that in C#, but it is possible in IL.

Also, I believe you may be losing the information on whether the type is actual primitive or enum.

Reply:
Ah, indeed, I've just added test cases for Enum and they fail. I am not sure yet how to check for "Is Enum" it seems it might require a new jit-vm API and in this case it will be probably too complicated.
Also, I am currently adding `ilproj` tests for the `ldtoken int&` case.